mmcls/models/backbones/levit.py

```diff
@@ -0,0 +1,545 @@
+            activation(), LinearBatchNorm(self.dh, dim, bn_weight_init=0))
+
+        points = list(itertools.product(range(resolution), range(resolution)))
+        N = len(points)
+        attention_offsets = {}
+        idxs = []
+        for p1 in points:
+            for p2 in points:
+                offset = (abs(p1[0] - p2[0]), abs(p1[1] - p2[1]))
+                if offset not in attention_offsets:
+                    attention_offsets[offset] = len(attention_offsets)
+                idxs.append(attention_offsets[offset])
+        self.attention_biases = torch.nn.Parameter(
+            torch.zeros(num_heads, len(attention_offsets)))
+        self.register_buffer('attention_bias_idxs',
+                             torch.LongTensor(idxs).view(N, N))
+
+    @torch.no_grad()
+    def train(self, mode=True):
+        super(Attention, self).train(mode)
+        if mode and hasattr(self, 'ab'):
+            del self.ab
+        else:
+            self.ab = self.attention_biases[:, self.attention_bias_idxs]
+
+    def forward(self, x):  # x (B,N,C)
+        B, N, C = x.shape  # 2 196 128
+        qkv = self.qkv(x)  # 2 196 128
+        q, k, v = qkv.view(B, N, self.num_heads, -1).split(
+            [self.key_dim, self.key_dim, self.d],
+            dim=3)  # q 2 196 4 16 ; k 2 196 4 16; v 2 196 4 32
+        q = q.permute(0, 2, 1, 3)  # 2 4 196 16
+        k = k.permute(0, 2, 1, 3)
+        v = v.permute(0, 2, 1, 3)
+
+        attn = ((q @ k.transpose(-2, -1)) *
+                self.scale  # 2 4 196 16 * 2 4 16 196 -> 2 4 196 196
+                + (self.attention_biases[:, self.attention_bias_idxs]
+                   if self.training else self.ab))
+        attn = attn.softmax(dim=-1)  # 2 4 196 196 -> 2 4 196 196
+        x = (attn @ v).transpose(1, 2).reshape(
+            B, N,
+            self.dh)  # 2 4 196 196 * 2 4 196 32 -> 2 4 196 32 -> 2 196 128
+        x = self.proj(x)
+        return x
+
+
+class MLP(nn.Sequential):
+
+    def __init__(
+        self,
+        embed_dim,
+        mlp_ratio,
+        mlp_activation,
+    ):
+        super(MLP, self).__init__()
+        h = embed_dim * mlp_ratio
+        self.linear1 = LinearBatchNorm(embed_dim, h)
+        self.activation = mlp_activation()
+        self.linear2 = LinearBatchNorm(h, embed_dim, bn_weight_init=0)
+
+    def forward(self, x):
+        x = self.linear1(x)
+        x = self.activation(x)
+        x = self.linear2(x)
+        return x
+
+
+class Subsample(BaseModule):
+
+    def __init__(self, stride, resolution):
+        super(Subsample, self).__init__()
+        self.stride = stride
+        self.resolution = resolution
+
+    def forward(self, x):
+        B, N, C = x.shape  # 2 196 128
+        x = x.view(B, self.resolution, self.resolution,
+                   C)[  # 2 196 128 -> 2 14 14 128  -> 2 7 7 128
+                       :, ::self.stride, ::self.stride].reshape(
+                           B, -1, C)  # # 2 14 14 128 > 2 49 128
```

Comment:
Better to use names instead of numbers in comments.
```suggestion
        B, _, C = x.shape
        # B, N, C -> B, H, W, C
        x = x.view(B, self.resolution, self.resolution, C)
        x = x[:, ::self.stride, ::self.stride]
        x = x.reshape(B, -1, C)  # B, H', W', C -> B, N', C
```